src/app/clan/clan-milestones/clan-milestones.component.html

```diff
@@ -76,7 +76,7 @@
                             <td *ngIf="(hiddenClanMilestones|async).indexOf(mileStoneName.key)===-1"
                                 class="clan-player-ms-cell" [attr.data-label]="mileStoneName.name">
                                 <ng-container *ngFor="let char of player.characters">
-                                    <d2c-milestone-check [char]="char" [mileStoneName]="mileStoneName" detailed="false">
+                                    <d2c-milestone-check [milestoneStatus]="char.milestones[mileStoneName.key]" detailed="false">
```

Comment:
I refactored the milestone-check component to take the milestoneStatus object that you currently have keyed in the character object under player. The reason being that the way I store my data, each row holds the status for each character, but doesn't have a reference to the character object or the original milestoneName object.